### ISSUE TITLE
Separate constraints and move to the end

### ISSUE THREAD

**Issue by oste** [michaelsogos/pg-diff-api#10]
I think it would be ideal to separate constraints and move those to the end. I ran into a 🐔 and 🥚 situation where a table was not available for a specified constraint. Moving constraints to the end as a final update should resolve this.

**Comment by patburdettgh** [comment]
agreed... FKs especially.... Thank you I love your stuff it works really good!

**Comment by michaelsogos** [comment]
Totally agree guys, sorry I plan for fix it.

**Comment by michaelsogos** [comment]
Hi @oste  @patburdettgh ,

I tried to investigate the case but could not replicate it.
Currently, constraints are always dropped at the beginning, before the alter table, in case they are needed, and re-created at the end of any changes concerning that specific table.

Regarding FK constraints, the patch should CREATE or ALTER the parent table before applying any changes to the child table containing the FK constraint (there is specific logic to sort CREATE/ALTER table commands). So, under my test, I cannot see issues like "patch create/change an FK constraint while the reference table/column does not exist yet; that should be the case of this issue.
In other terms could you supply me an example of a situation to replicate to debug my work?

**Comment by targhs** [comment]
+1 to the issue. I faced it too, But can't provide the db structure from my application. Will try replicating it in a dummy db
